### ISSUE TITLE
Inject HOC decorator only injects properties from class stores without any methods.

### ISSUE THREAD

**Issue by marcus-sa** [mobxjs/mobx-react#518]
From how I've used `mobx-react` in the past, it should inject class methods (`actions`) aswell.
But for some reason it only seems to inject class properties (`observables`)

[CodeSandbox Example](https://codesandbox.io/s/l4o3y8n3zl)

**Comment by urugator** [comment]
`@inject(({ test }) => test)`
Inject is supposed to return a set of props. Inject uses `for in` loop to merge the returned props with original set of props. However `@action` and `@computed` props are not enumerable, therefore ignored by `for in` loop. 
Also note that `this` would be `undefined` during `await this.props.fetch()` call, because the action is not bound to store instance.

**Comment by mweststrate** [comment]
If you want to inject an class instance, just inject the entire object instance, rather than trying to destruct it into the props of the injected component. What you are doing here is very confusing, because `props` would not be a plain object anymore, but a kind of half-arsed class instance (with all the members, but not the prototype or the correct `this` binding). `this.props` will never be a `TestStore` instance, but you are treating like it is one, by calling `fetch` on it, and thereby probably breaking all the assumptions `fetch` might have from the fact that it was defined in the `TestStore` class. For example by tring to make `super` calls. 

 In other words, just leave `test` as a whole thing and write `this.props.test.fetch()`.

**Comment by michaelspeed** [comment]
i suppose is there a work around?

**Comment by FredyC** [comment]
@michaelspeed Please open the new issue if you have more details about what you are solving. This issue has been closed a long time ago.
